test/gtest/uct/test_p2p_mix.cc

```diff
@@ -200,8 +201,17 @@ void uct_p2p_mix_test::run(unsigned count, size_t offset, size_t size_cap)
     mapped_buffer sendbuf = alloc_buffer(sender(), offset);
     mapped_buffer recvbuf = alloc_buffer(receiver(), offset);
 
+    auto start_time = ucs_get_accurate_time();
     for (unsigned i = 0; i < count; ++i) {
         random_op(sendbuf, recvbuf);
+
+        /* Some transports may send larger buffers or can be slower, so limit
+           also by time */
+        if (ucs_get_accurate_time() > (start_time + timeout)) {
+            UCS_TEST_MESSAGE << "Stopped after " << i
+                             << " iterations due to time limit";
+            break;
+        }
```

Comment:
is it unrelated fix?

Reply:
it is needed because cuda_copy tests are being run with up to 1GB buffer now and take much more time to complete